diff --git a/src/Feeds/Fetcher/HttpFetcher.php b/src/Feeds/Fetcher/HttpFetcher.php
index b73e669..e7f102c 100644
--- a/src/Feeds/Fetcher/HttpFetcher.php
+++ b/src/Feeds/Fetcher/HttpFetcher.php
@@ -7,6 +7,7 @@
 namespace Drupal\feeds\Feeds\Fetcher;
 
 use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
 use Drupal\Core\File\FileSystemInterface;
 use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
 use Drupal\feeds\Exception\EmptyFeedException;
@@ -40,6 +41,8 @@
  * )
  */
 class HttpFetcher extends PluginBase implements ClearableInterface, FetcherInterface, ContainerFactoryPluginInterface {
+  use DependencySerializationTrait;
+
   /**
    * The Guzzle client.
    *
diff --git a/src/Feeds/State/CleanState.php b/src/Feeds/State/CleanState.php
index f56e210..e7f102c 100644
--- a/src/Feeds/State/CleanState.php
+++ b/src/Feeds/State/CleanState.php
@@ -106,6 +106,31 @@
   /**
    * {@inheritdoc}
    */
+  public function __sleep() {
+    $vars = $this->dependencySerializationTraitSleep();
+    foreach (['connection', 'entityTypeManager'] as $prop) {
+      if (($key = array_search($prop, $vars)) !== FALSE) {
+        unset($vars[$key]);
+      }
+    }
+    return $vars;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __wakeup() {
+    $this->dependencySerializationTraitWakeUp();
+    $container = \Drupal::getContainer();
+    $this->connection = $container->get('database');
+    $this->entityTypeManager = $container->get('entity_type.manager');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   public function progress($total, $progress) {
     if (!$this->count()) {
       $this->setCompleted();
diff --git a/src/State.php b/src/State.php
index 3a2c5e1..e7f102c 100644
--- a/src/State.php
+++ b/src/State.php
@@ -145,10 +145,11 @@
   public function __sleep() {
     $vars = $this->dependencySerializationTraitSleep();
 
-    // Do not serialize the logger object.
-    $key = array_search('logger', $vars);
-    if ($key !== FALSE) {
-      unset($vars[$key]);
+    // Do not serialize the logger and messenger objects.
+    foreach (['logger', 'messenger'] as $prop) {
+      if (($key = array_search($prop, $vars)) !== FALSE) {
+        unset($vars[$key]);
+      }
     }
 
     return $vars;
