diff --git a/src/Feeds/Fetcher/HttpFetcher.php b/src/Feeds/Fetcher/HttpFetcher.php
index c4ad512..8d90cfb 100644
--- a/src/Feeds/Fetcher/HttpFetcher.php
+++ b/src/Feeds/Fetcher/HttpFetcher.php
@@ -3,6 +3,7 @@
 namespace Drupal\feeds\Feeds\Fetcher;
 
 use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
 use Drupal\Core\File\FileSystemInterface;
 use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
 use Drupal\feeds\Exception\EmptyFeedException;
@@ -36,6 +37,8 @@ use Symfony\Component\HttpFoundation\Response;
  */
 class HttpFetcher extends PluginBase implements ClearableInterface, FetcherInterface, ContainerFactoryPluginInterface {
 
+  use DependencySerializationTrait;
+
   /**
    * The Guzzle client.
    *
diff --git a/src/Feeds/State/CleanState.php b/src/Feeds/State/CleanState.php
index 56afa34..c010dc3 100644
--- a/src/Feeds/State/CleanState.php
+++ b/src/Feeds/State/CleanState.php
@@ -100,6 +100,29 @@ class CleanState extends State implements CleanStateInterface {
     );
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function __sleep() {
+    $vars = $this->dependencySerializationTraitSleep();
+    foreach (['connection', 'entityTypeManager'] as $prop) {
+      if (($key = array_search($prop, $vars)) !== FALSE) {
+        unset($vars[$key]);
+      }
+    }
+    return $vars;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __wakeup() {
+    $this->dependencySerializationTraitWakeUp();
+    $container = \Drupal::getContainer();
+    $this->connection = $container->get('database');
+    $this->entityTypeManager = $container->get('entity_type.manager');
+  }
+
   /**
    * {@inheritdoc}
    */
diff --git a/src/State.php b/src/State.php
index c52fbc0..1a2498f 100644
--- a/src/State.php
+++ b/src/State.php
@@ -141,10 +141,11 @@ class State implements StateInterface, ContainerStateInterface {
   public function __sleep() {
     $vars = $this->dependencySerializationTraitSleep();
 
-    // Do not serialize the logger object.
-    $key = array_search('logger', $vars);
-    if ($key !== FALSE) {
-      unset($vars[$key]);
+    // Do not serialize the logger and messenger objects.
+    foreach (['logger', 'messenger'] as $prop) {
+      if (($key = array_search($prop, $vars)) !== FALSE) {
+        unset($vars[$key]);
+      }
     }
 
     return $vars;
